### PR TITLE
GooFit package

### PR DESCRIPTION
**Pull request by henryiii**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Starting a CPU/GPU package for GooFit. Will need some work, probably, unless I'm very lucky (and regardless, I can add tests).

### CODE REVIEW

File: recipes/goofit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - {{ compiler("cuda") }}  # [linux and cuda_compiler_version != "None"]
+        - cmake
+      host:
+        - python
+        - scikit-build
+        - pip
+        - llvm-openmp  # [c_compiler == "clang"]
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp

File: recipes/goofit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      run:
+        - python
+        - numpy>=1.11.1
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]:
```suggestion
        - numpy >=1.11.1
```

File: recipes/goofit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      source_files:
+        - python/tests/test_exp.py
+      imports:
+        - goofit
+      commands:
+        - $PYTHON -m pytest python/tests/test_exp.py
+
+    about:
+      home: https://github.com/GooFit/GooFit
+      license: LGPL-3
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: It should be one of these:
```suggestion
      license: LGPL-3.0-only
```

```suggestion
      license: LGPL-3.0-or-later
```

File: recipes/goofit/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      requires:
+        - pytest
+      source_files:
+        - python/tests/test_exp.py
+      imports:
+        - goofit
+      commands:
+        - $PYTHON -m pytest python/tests/test_exp.py
+
+    about:
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: This metadata can be inherited from the top level section

Comment (henryiii) [reply recipes/goofit/meta.yaml]: I was following ucx-split-feedstock exactly, which listed it twice.

File: recipes/goofit/meta.yaml (hunk 1 above)
Comment (henryiii) [reply recipes/goofit/meta.yaml]: Thanks! I was looking for that but couldn't find it.

**Comment by henryiii** [comment]
@danielsibemol and @chrisburr, please verify you are okay being listed as maintainers.

**Comment by henryiii** [comment]
PS: I'm not sure the CUDA build gets tested in staged recipes, I don't see it listed.

**Comment by henryiii** [comment]
I'm going to test here: https://github.com/conda-forge/root-feedstock/pull/77

**Comment by danielsibemol** [comment]
> @danielsibemol and @chrisburr, please verify you are okay being listed as maintainers.

Sure! I am ok with being maintainer.

**Comment by henryiii** [comment]
@chrisburr, I've tested this here: https://github.com/conda-forge/root-feedstock/pull/77

macOS will fail due to 10.9 being used instead of 10.12, and the CUDA part is not tested in staged-recipes.

File: recipes/goofit/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 82884fd3021555da0953776f81199cbea60287f6bca87b0ecdaa5dd212692c67
+  patches:
+    - patches/thrustorder-2.2.2.patch
+  
+build:
+  skip: true  # [win]
+  {% if cuda_compiler_version|default("None") in goofit_cuda_supported %}
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: `in` -> `not in` and drop the else? Or is it needed to work around a rendering quirk?

File: recipes/goofit/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        - libgomp      # [linux and not aarch64]
+      host:
+        - python
+        - scikit-build
+        - pip
+        - llvm-openmp  # [osx]
+        - libgomp      # [linux and not aarch64]
+      run:
+        - python
+        - numpy >=1.11.1
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: Are you linking against numpy's C API? If you aren't, this is fine.

File: recipes/goofit/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    test:
+      requires:
+        - pytest  # [not linux or cuda_compiler_version == "None"]
+      source_files:
+        - python/tests/test_exp.py  # [not linux or cuda_compiler_version == "None"]
+      imports:
+        - goofit  # [not linux or cuda_compiler_version == "None"]
+      commands:
+        - $PYTHON -m pytest python/tests/test_exp.py  # [not linux or cuda_compiler_version == "None"]
+        - $PYTHON -c "import sys, pkgutil; sys.exit(pkgutil.find_loader('goofit') is None)"
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: Nice! 😆

File: recipes/goofit/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      number: 0
+      string: "{{ goofit_proc_type }}"
+    test:
+      commands:
+        - exit 0
+    about:
+      home: https://github.com/conda-forge/goofit-feedstock
+      license: BSD-3-Clause
+      license_family: BSD
+      summary: A meta-package to select CPU or GPU GooFit build.
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: It's really minor but add whitespace so it's easier to mentally separate the outputs:
```suggestion
      summary: A meta-package to select CPU or GPU GooFit build.

```

File: recipes/goofit/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,91 @@
+{% set version = "2.2.2" %}
+{% set name = "goofit" %}
+
+{% set goofit_proc_version = "1.0.0" %}
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]: As it's only used in one place I wouldn't bother with the variable:
```suggestion
```

File: recipes/goofit/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  skip: true  # [win]
+  {% if cuda_compiler_version|default("None") in goofit_cuda_supported %}
+  {% else %}
+  skip: true
+  {% endif %}
+  number: 0
+
+outputs:
+  - name: goofit-proc
+    version: "{{ goofit_proc_version }}"
```
Comment (chrisburr) [inline recipes/goofit/meta.yaml]:
```suggestion
    version: "1.0.0"
```

File: recipes/goofit/meta.yaml (hunk 5 above)
Comment (henryiii) [reply recipes/goofit/meta.yaml]: I thought so, but it seems to work now, so must have had a different issue earlier. Updated.

File: recipes/goofit/meta.yaml (hunk 6 above)
Comment (henryiii) [reply recipes/goofit/meta.yaml]: Nope, it's PyBind11, which does not link to numpy like Cython does.

**Pull request merged**